README.md
# PatientNama
A Project build on python. This project was made to manage a patient data and generates a pdf file to produce a discharge statement.

Choosing a database: Choose a database to store the patient data. Data on this stored on an excel file.

Set up your environment: Install Python on your computer and set up a virtual environment to manage your project dependencies.

Create a patient model: Create a patient model to represent the patient data in your database. This model should include fields for the patient's name, age, gender, medical history, and any other relevant information.

Build a user interface: A GUI toolkit like PyQt to build a user interface for your project. This interface should allow users to enter and edit patient data, view patient records, and generate a discharge statement in PDF format.

Generate a PDF file: A PDF library like ReportLab or PyPDF2 to generate a PDF file from the patient data. This file should include the patient's name, age, gender, medical history, and any other relevant information.

Test and debug: Tested the project to make sure it works as intended. You can use unit tests and integration tests to ensure that your code is functioning correctly. You should also debug any issues that arise.

Deploy your project: Once your project is complete, create a standalone executable using tools like PyInstaller.

Extract The All Libraries Folder's Contents in the main Folder to run.
To Run Install the relavent Libraries and Run PatientName-Run Application File.


NEWS.txt
+++++++++++
Python News
+++++++++++

What's New in Python 3.11.3 final?
==================================

*Release date: 2023-04-04*

Security
--------

- gh-issue-101727: Updated the OpenSSL version used in Windows and macOS
  binary release builds to 1.1.1t to address CVE-2023-0286, CVE-2022-4303,
  and CVE-2022-4303 per `the OpenSSL 2023-02-07 security advisory
  <https://www.openssl.org/news/secadv/20230207.txt>`_.

- gh-issue-101283: :class:`subprocess.Popen` now uses a safer approach to
  find ``cmd.exe`` when launching with ``shell=True``. Patch by Eryk Sun,
  based on a patch by Oleg Iarygin.

Core and Builtins
-----------------

- gh-issue-101975: Fixed ``stacktop`` value on tracing entries to avoid
  corruption on garbage collection.

- gh-issue-102701: Fix overflow when creating very large dict.

- gh-issue-102416: Do not memoize incorrectly automatically generated loop
  rules in the parser. Patch by Pablo Galindo.

- gh-issue-102356: Fix a bug that caused a crash when deallocating deeply
  nested filter objects. Patch by Marta Gómez Macías.

- gh-issue-102397: Fix segfault from race condition in signal handling
  during garbage collection. Patch by Kumar Aditya.

- gh-issue-102281: Fix potential nullptr dereference and use of
  uninitialized memory in fileutils. Patch by Max Bachmann.

- gh-issue-102126: Fix deadlock at shutdown when clearing thread states if
  any finalizer tries to acquire the runtime head lock. Patch by Kumar
  Aditya.

- gh-issue-102027: Fix SSE2 and SSE3 detection in ``_blake2`` internal
  module. Patch by Max Bachmann.

- gh-issue-101967: Fix possible segfault in
  ``positional_only_passed_as_keyword`` function, when new list created.

- gh-issue-101765: Fix SystemError / segmentation fault in iter
  ``__reduce__`` when internal access of ``builtins.__dict__`` keys mutates
  the iter object.

- gh-issue-101696: Invalidate type version tag in ``_PyStaticType_Dealloc``
  for static types, avoiding bug where a false cache hit could crash the
  interpreter. Patch by Kumar Aditya.

Library
-------

- gh-issue-102549: Don't ignore exceptions in member type creation.

- gh-issue-102947: Improve traceback when :func:`dataclasses.fields` is
  called on a non-dataclass. Patch by Alex Waygood

- gh-issue-102780: The :class:`asyncio.Timeout` context manager now works
  reliably even when performing cleanup due to task cancellation.
  Previously it could raise a :exc:`~asyncio.CancelledError` instead of an
  :exc:`~asyncio.TimeoutError` in such cases.

- gh-issue-88965: typing: Fix a bug relating to substitution in custom
  classes generic over a :class:`~typing.ParamSpec`. Previously, if the
  ``ParamSpec`` was substituted with a parameters list that itself contained
  a :class:`~typing.TypeVar`, the ``TypeVar`` in the parameters list could
  not be subsequently substituted. This is now fixed.

  Patch by Nikita Sobolev.

- gh-issue-101979: Fix a bug where parentheses in the ``metavar`` argument
  to :meth:`argparse.ArgumentParser.add_argument` were dropped. Patch by
  Yeojin Kim.

- gh-issue-102179: Fix :func:`os.dup2` error message for negative fds.

- gh-issue-101961: For the binary mode, :func:`fileinput.hookcompressed`
  doesn't set the ``encoding`` value even if the value is ``None``. Patch by
  Gihwan Kim.

- gh-issue-101936: The default value of ``fp`` becomes :class:`io.BytesIO`
  if :exc:`~urllib.error.HTTPError` is initialized without a designated
  ``fp`` parameter. Patch by Long Vo.

- gh-issue-102069: Fix ``__weakref__`` descriptor generation for custom
  dataclasses.

- gh-issue-101566: In zipfile, apply fix for extractall on the underlying
  zipfile after being wrapped in ``Path``.

- gh-issue-101892: Callable iterators no longer raise :class:`SystemError`
  when the callable object exhausts the iterator but forgets to either
  return a sentinel value or raise :class:`StopIteration`.

- gh-issue-97786: Fix potential undefined behaviour in corner cases of
  floating-point-to-time conversions.

- gh-issue-101517: Fixed bug where :mod:`bdb` looks up the source line with
  :mod:`linecache` with a ``lineno=None``, which causes it to fail with an
  unhandled exception.

- gh-issue-101673: Fix a :mod:`pdb` bug where ``ll`` clears the changes to
  local variables.

- gh-issue-96931: Fix incorrect results from
  :meth:`ssl.SSLSocket.shared_ciphers`

- gh-issue-88233: Correctly preserve "extra" fields in ``zipfile``
  regardless of their ordering relative to a zip64 "extra."

- gh-issue-96127: ``inspect.signature`` was raising ``TypeError`` on call
  with mock objects. Now it correctly returns ``(*args, **kwargs)`` as
  infered signature.

- gh-issue-95495: When built against OpenSSL 3.0, the :mod:`ssl` module had
  a bug where it reported unauthenticated EOFs (i.e. without close_notify)
  as a clean TLS-level EOF. It now raises :exc:`~ssl.SSLEOFError`, matching
  the behavior in previous versions of OpenSSL. The
  :attr:`~ssl.SSLContext.options` attribute on :class:`~ssl.SSLContext` also
  no longer includes :data:`~ssl.OP_IGNORE_UNEXPECTED_EOF` by default. This
  option may be set to specify the previous OpenSSL 3.0 behavior.

- gh-issue-94440: Fix a :mod:`concurrent.futures.process` bug where
  ``ProcessPoolExecutor`` shutdown could hang after a future has been
  quickly submitted and canceled.

Documentation
-------------

- gh-issue-103112: Add docstring to :meth:`http.client.HTTPResponse.read` to
  fix ``pydoc`` output.

- gh-issue-85417: Update :mod:`cmath` documentation to clarify behaviour on
  branch cuts.

- gh-issue-97725: Fix :meth:`asyncio.Task.print_stack` description for
  ``file=None``. Patch by Oleg Iarygin.

Tests
-----

- gh-issue-102980: Improve test coverage on :mod:`pdb`.

- gh-issue-102537: Adjust the error handling strategy in

test.py
import PyQt5
import sys
from pathlib import Path
from PyQt5 import QtCore, QtGui, QtWidgets
from PyQt5.QtWidgets import *
from PyQt5.QtGui import *
from PyQt5.QtCore import *
import os
import pandas as pd
import time as t
from reportlab.lib.pagesizes import A4, letter
from reportlab.pdfgen import canvas
import numpy as np
from datetime import date
from PyPDF2 import PdfMerger


def pdfGenerate(wordList):
    if 1:
        y = os.environ.get('USERNAME')
        my_canvas = canvas.Canvas(
            "C:\\Users\\"+str(y)+"\\Downloads\\"+wordList[0]+"_"+wordList[6]+"(0).pdf", pagesize=A4)
        my_canvas.setLineWidth(.3)
        my_canvas.setFont('Helvetica', 15)
        my_canvas.drawImage("Me.jpg", 30, 770, width=60, height=60)
        my_canvas.drawString(
            100, 800, "Sheikh Zayad Hospital-Surgical Unit-l, Lahore")

        my_canvas.setFont('Helvetica', 12)
        my_canvas.drawString(470, 780, 'Discharge Summary')
        my_canvas.setFont('Helvetica', 11)
        my_canvas.drawString(420, 730, "Discharge Date: ")
        today = date.today()
        d2 = today.strftime("%B %d, %Y")
        my_canvas.drawString(510, 730, d2)
        my_canvas.line(460, 773, 580, 773)

        my_canvas.drawString(440, 750, 'Admit Date:')
        my_canvas.drawString(510, 750, wordList[59])

        my_canvas.drawString(350, 710, 'Consultant Name:')
        my_canvas.drawString(450, 710, wordList[60])
        my_canvas.drawString(350, 710, 'Consultant Name:')
        my_canvas.drawString(450, 710, wordList[60])
# ----------------------------------------------------------------------------------------------
        my_canvas.setFont('Helvetica', 12)
        my_canvas.drawString(30, 680, 'Patient Information:')
        my_canvas.line(30, 670, 580, 670)
        my_canvas.setFont('Helvetica', 12)
        my_canvas.drawString(30, 680, 'Patient Information:')
        my_canvas.line(30, 670, 580, 670)

        my_canvas.setFont('Helvetica', 10)
        my_canvas.drawString(40, 650, 'Name:')
        my_canvas.drawString(100, 650, wordList[0])
        my_canvas.drawString(40, 630, 'Gender:')
        my_canvas.drawString(100, 630, wordList[1])
        my_canvas.drawString(40, 610, 'Address:')
        my_canvas.drawString(100, 610, wordList[2])

        my_canvas.drawString(250, 650, 'Age:')
        my_canvas.drawString(280, 650, wordList[3])
        my_canvas.drawString(250, 630, 'Phone No:')
        my_canvas.drawString(315, 630, wordList[4])
        my_canvas.drawString(250, 610, 'City:')
        my_canvas.drawString(280, 610, wordList[5])

        my_canvas.drawString(430, 650, 'Cnic:')
        my_canvas.drawString(460, 650, wordList[6])
        my_canvas.drawString(430, 630, 'Hospital Number:')
        my_canvas.drawString(520, 630, wordList[7])

# ------------------------------------------------------------------------------------
        my_canvas.setFont('Helvetica', 12)
        my_canvas.drawString(30, 510+70, 'Clinical Data:')
        my_canvas.line(30, 500+70, 580, 500+70)
        my_canvas.setFont('Helvetica', 12)
        my_canvas.drawString(30, 510+70, 'Clinical Data:')
        my_canvas.line(30, 500+70, 580, 500+70)

        my_canvas.setFont('Helvetica', 10)
        my_canvas.drawString(50, 475+70, 'Present Complaint')
        my_canvas.setFont('Helvetica', 9.5)
        my_canvas.drawString(50, 460+70, wordList[8])
        # my_canvas.rect(40, 440, 530, 120, stroke=1, fill=0)

        my_canvas.setFont('Helvetica', 10)
        my_canvas.drawString(50, 420+70-50, 'Examination Findings')
        my_canvas.setFont('Helvetica', 9.5)
        my_canvas.drawString(50, 400+70-50, wordList[9])
        # my_canvas.rect(310, 430, 220, 100, stroke=1, fill=0)
# -------------------------------------------------------------------------------------
        my_canvas.setFont('Helvetica', 12)
        my_canvas.drawString(30, 420-100, 'Investigations:')
        my_canvas.line(30, 410-100, 580, 410-100)
        my_canvas.setFont('Helvetica', 12)
        my_canvas.drawString(30, 420-100, 'Investigations:')
        my_canvas.line(30, 410-100, 580, 410-100)

        my_canvas.setFont('Helvetica', 10)
        my_canvas.drawString(30, 390-100, 'CBC:')
        my_canvas.drawString(100, 390-100, wordList[10])
        my_canvas.drawString(30, 375-100, 'LFT:')
        my_canvas.drawString(100, 375-100, wordList[11])
        my_canvas.drawString(30, 360-100, 'Electrolyte:')
        my_canvas.drawString(100, 360-100, wordList[12])
        my_canvas.drawString(30, 345-100, 'Viral Markers:')
        my_canvas.drawString(100, 345-100, wordList[13])
        my_canvas.drawString(30, 330-100, 'Imaging:')
        my_canvas.drawString(100, 330-100, wordList[14])
        my_canvas.drawString(30, 295+20-100, 'Others:')
        my_canvas.drawString(100, 295+20-100, wordList[15])

        my_canvas.setFont('Helvetica', 12)
        my_canvas.drawString(30, 285-100, 'Operative Notes')
        my_canvas.setFont('Helvetica', 12)

        my_canvas.setFont('Helvetica', 10)
        my_canvas.drawString(40, 265-100, wordList[16])
 # -----------------------------------------------------------------------------
        my_canvas.line(30, 280-100, 580, 280-100)
        my_canvas.save()
# ----------------------------------------------------------------------------------------------
        my_canvasT = canvas.Canvas(
            "C:\\Users\\"+str(y)+"\\Downloads\\"+wordList[0]+"_"+wordList[6]+"(1).pdf", pagesize=A4)
        my_canvasT.setLineWidth(.3)
        my_canvasT.setFont('Helvetica', 15)
        my_canvasT.drawImage("Me.jpg", 30, 770, width=60, height=60)
        my_canvasT.drawString(
            100, 800, "Sheikh Zayad Hospital-Surgical Unit-l, Lahore")

        my_canvasT.setFont('Helvetica', 12)
        my_canvasT.drawString(470, 780, 'Discharge Summary')
        my_canvasT.setFont('Helvetica', 11)
        my_canvasT.drawString(420, 730, "Discharge Date: ")
        today = date.today()
        d2 = today.strftime("%B %d, %Y")
        my_canvasT.drawString(510, 730, d2)
        my_canvasT.line(460, 773, 580, 773)

        my_canvasT.drawString(440, 750, 'Admit Date:')
        my_canvasT.drawString(510, 750, wordList[59])

        my_canvasT.drawString(350, 710, 'Consultant Name:')
        my_canvasT.drawString(450, 710, wordList[60])
        my_canvasT.drawString(350, 710, 'Consultant Name:')
        my_canvasT.drawString(450, 710, wordList[60])

        my_canvasT.drawString(30, 680, 'Course / Treatment')
        my_canvasT.line(30, 675, 580, 675)

        my_canvasT.setFont('Helvetica', 9.5)
        my_canvasT.drawString(40, 660, wordList[17])
        # my_canvas.rect(310, 210, 220, 50, stroke=1, fill=0)
# ------------------------------------------------------------------------------
        my_canvasT.setFont('Helvetica', 13)
        my_canvasT.drawString(30, 198+300, 'Discharge Medicines')

        my_canvasT.setFont('Helvetica', 13)
        my_canvasT.drawString(30, 40+300, 'FollowUp Instructions')
        my_canvasT.line(30, 195+300, 580, 195+300)
        my_canvasT.line(30, 35+300, 580, 35+300)

        my_canvasT.setFont('Helvetica', 11)
        my_canvasT.drawString(40, 175+300, "Name")
        my_canvasT.drawString(150, 175+300, "Dose")
        my_canvasT.drawString(250, 175+300, "Times")
        my_canvasT.drawString(350, 175+300, "Days")

        my_canvasT.setFont('Helvetica', 9)
        my_canvasT.drawString(40, 160+300, wordList[19])
        my_canvasT.drawString(150, 160+300, wordList[20])
        my_canvasT.drawString(250, 160+300, wordList[21])
        my_canvasT.drawString(350, 160+300, wordList[22])

        my_canvasT.setFont('Helvetica', 9)
        my_canvasT.drawString(40, 150+300, wordList[23])
        my_canvasT.drawString(150, 150+300, wordList[24])
        my_canvasT.drawString(250, 150+300, wordList[25])
        my_canvasT.drawString(350, 150+300, wordList[26])

        my_canvasT.setFont('Helvetica', 9)
        my_canvasT.drawString(40, 140+300, wordList[27])
        my_canvasT.drawString(150, 140+300, wordList[28])
        my_canvasT.drawString(250, 140+300, wordList[29])
        my_canvasT.drawString(350, 140+300, wordList[30])

        my_canvasT.setFont('Helvetica', 9)
        my_canvasT.drawString(40, 130+300, wordList[31])
        my_canvasT.drawString(150, 130+300, wordList[32])
        my_canvasT.drawString(250, 130+300, wordList[33])
        my_canvasT.drawString(350, 130+300, wordList[34])

        my_canvasT.setFont('Helvetica', 9)
        my_canvasT.drawString(40, 120+300, wordList[35])
        my_canvasT.drawString(150, 120+300, wordList[36])
        my_canvasT.drawString(250, 120+300, wordList[37])
        my_canvasT.drawString(350, 120+300, wordList[38])

        my_canvasT.setFont('Helvetica', 9)
        my_canvasT.drawString(40, 110+300, wordList[39])
        my_canvasT.drawString(150, 110+300, wordList[40])
        my_canvasT.drawString(250, 110+300, wordList[41])
        my_canvasT.drawString(350, 110+300, wordList[42])

        my_canvasT.setFont('Helvetica', 9)
        my_canvasT.drawString(40, 100+300, wordList[43])
        my_canvasT.drawString(150, 100+300, wordList[44])
        my_canvasT.drawString(250, 100+300, wordList[45])
        my_canvasT.drawString(350, 100+300, wordList[46])

        my_canvasT.setFont('Helvetica', 9)
        my_canvasT.drawString(40, 90+300, wordList[47])
        my_canvasT.drawString(150, 90+300, wordList[48])
        my_canvasT.drawString(250, 90+300, wordList[49])
        my_canvasT.drawString(350, 90+300, wordList[50])

        my_canvasT.setFont('Helvetica', 9)
        my_canvasT.drawString(40, 80+300, wordList[51])
        my_canvasT.drawString(150, 80+300, wordList[52])
        my_canvasT.drawString(250, 80+300, wordList[53])
        my_canvasT.drawString(350, 80+300, wordList[54])

        my_canvasT.setFont('Helvetica', 9)
        my_canvasT.drawString(40, 70+300, wordList[55])
        my_canvasT.drawString(150, 70+300, wordList[56])
        my_canvasT.drawString(250, 70+300, wordList[57])
        my_canvasT.drawString(350, 70+300, wordList[58])

        my_canvasT.setFont('Helvetica', 9)
        # my_canvasT.rect(30, 60+300, 550, 135, stroke=1, fill=0)
        my_canvasT.drawString(40, 25+300, wordList[18])

        my_canvasT.setFont('Helvetica', 9)
        my_canvasT.drawString(
            50, 10+100, "Doctor's Signature:_____________________")

        my_canvasT.setFont('Helvetica', 9)
        my_canvasT.drawString(
            330, 10+100, "Consultant's Signature:_____________________")

        my_canvasT.save()
# ----------------------------------------------------------------------------------------------

        # Create an instance of PdfFileMerger() class
        merger = PdfMerger()

        # Create a list with the file paths
        pdf_files = ["C:\\Users\\"+str(y)+"\\Downloads\\"+wordList[0]+"_"+wordList[6]+"(0).pdf",
                     "C:\\Users\\"+str(y)+"\\Downloads\\"+wordList[0]+"_"+wordList[6]+"(1).pdf"]

        # Iterate over the list of the file paths
        for pdf_file in pdf_files:
            # Append PDF files
            merger.append(pdf_file)

        # Write out the merged PDF file
        merger.write("C:\\Users\\"+str(y)+"\\Downloads\\" +
                     wordList[0]+"_"+wordList[6]+".pdf")
        merger.close()

        if os.path.exists("C:\\Users\\"+str(y)+"\\Downloads\\"+wordList[0]+"_"+wordList[6]+"(0).pdf"):
            os.remove("C:\\Users\\"+str(y)+"\\Downloads\\" +
                      wordList[0]+"_"+wordList[6]+"(0).pdf")
        else:
            print("The file does not exist")

        if os.path.exists("C:\\Users\\"+str(y)+"\\Downloads\\"+wordList[0]+"_"+wordList[6]+"(1).pdf"):
            os.remove("C:\\Users\\"+str(y)+"\\Downloads\\" +
                      wordList[0]+"_"+wordList[6]+"(1).pdf")
        else:
            print("The file does not exist")

        os. system("C:\\Users\\"+str(y)+"\\Downloads\\" +
                   wordList[0]+"_"+wordList[6]+".pdf")


word = []
for i in range(61):
    word.append("")
pdfGenerate(word)


demo.py
import sys
import PyQt5
from PyQt5 import QtWidgets
from PyQt5.QtWidgets import *
from PyQt5.QtGui import QGuiApplication, QFont, QIcon
from PyQt5.QtCore import Qt
from PyQt5.QtCore import *
from PyQt5.QtGui import *

def window():
   app = QApplication(sys.argv)
   win = QWidget()

   l1 = QLabel("Name")
   nm = QLineEdit()

   l2 = QLabel("Address")
   add1 = QLineEdit()
   add2 = QLineEdit()
   fbox = QFormLayout()
   fbox.addRow(l1,nm)
   vbox = QVBoxLayout()

   vbox.addWidget(add1)
   vbox.addWidget(add2)
   fbox.addRow(l2,vbox)
   hbox = QHBoxLayout()

   r1 = QRadioButton("Male")
   r2 = QRadioButton("Female")
   hbox.addWidget(r1)
   hbox.addWidget(r2)
   hbox.addStretch()
   fbox.addRow(QLabel("sex"),hbox)
   fbox.addRow(QPushButton("Submit"),QPushButton("Cancel"))


   win.setLayout(fbox)
   
   win.setWindowTitle("PyQt")
   win.show()
   sys.exit(app.exec_())

if __name__ == '__main__':
   window()

extras.py
'''elif Patient_Present_Complaint == " |  | ":
            label.setStyleSheet("background-color: red")
            label.setText("Present Complaint Empty")
            label.move(1210, 700)
            label.resize(200, 30)
        elif Patient_Examination_Findings == " |  | ":
            label.setStyleSheet("background-color: red")
            label.setText("Exampination Findings Empty")
            label.move(1210, 700)
            label.resize(200, 30)
        elif Patient_Ivestigations_CBC == "":
            label.setStyleSheet("background-color: red")
            label.setText("CBC Empty")
            label.move(1210, 700)
            label.resize(200, 30)
        elif Patient_Ivestigations_LFT == "":
            label.setStyleSheet("background-color: red")
            label.setText("LFT Empty")
            label.move(1210, 700)
            label.resize(200, 30)
        elif Patient_Ivestigations_Electrolyte == "":
            label.setStyleSheet("background-color: red")
            label.setText("Electrolyte Empty")
            label.move(1210, 700)
            label.resize(200, 30)
        elif Patient_Ivestigations_Viral_Markers == "":
            label.setStyleSheet("background-color: red")
            label.setText("Viral Markers Empty")
            label.move(1210, 700)
            label.resize(200, 30)
        elif Patient_Ivestigations_Imaging == "":
            label.setStyleSheet("background-color: red")
            label.setText("Imaging Empty")
            label.move(1210, 700)
            label.resize(200, 30)
        elif Patient_Ivestigations_Others == "":
            label.setStyleSheet("background-color: red")
            label.setText("Others Empty")
            label.move(1210, 700)
            label.resize(200, 30)
        elif Patient_Operative_Notes == " |  | ":
            label.setStyleSheet("background-color: red")
            label.setText("Operative Notes Empty")
            label.move(1210, 700)
            label.resize(200, 30)
        elif Patient_Course_Treatment == " |  | ":
            label.setStyleSheet("background-color: red")
            label.setText("Course Treatment Empty")
            label.move(1210, 700)
            label.resize(200, 30)
        elif Patient_Follow_Up_Instructions == " |  | ":
            label.setStyleSheet("background-color: red")
            label.setText("Follow-Up Instructions Empty")
            label.move(1210, 700)
            label.resize(200, 30)
        '''

zipextimporter.py
r"""zipextimporter - an importer which can import extension modules
from zipfiles without unpacking them to the file system.

This file and _memimporter.pyd is part of the py2exe package.

Overview
========

zipextimporter.py contains the ZipExtImporter class which allows to
load Python binary extension modules contained in a zip.archive,
without unpacking them to the file system.

Call the zipextimporter.install() function to install the import hook,
add a zip-file containing .pyd or .dll extension modules to sys.path,
and import them.

It uses the _memimporter extension which uses code from Joachim
Bauch's MemoryModule library.  This library emulates the win32 api
function LoadLibrary.

Sample usage
============

You have to prepare a zip-archive 'lib.zip' containing
your Python's _socket.pyd for this example to work.

>>> import zipextimporter
>>> zipextimporter.install()
>>> import sys
>>> sys.path.insert(0, "lib.zip")
>>> import _socket
>>> print(_socket)
<module '_socket' from 'lib.zip\_socket.pyd'>
>>> _socket.__file__
'lib.zip\\_socket.pyd'
>>> _socket.__loader__
<ZipExtensionImporter object 'lib.zip'>
>>> # Reloading also works correctly:
>>> _socket is reload(_socket)
True
>>>

"""
import sys
import zipimport

from importlib import machinery
from importlib import util

# _memimporter is a module built into the py2exe runstubs.
import _memimporter

class ZipExtensionImporter(zipimport.zipimporter):
    _suffixes = machinery.EXTENSION_SUFFIXES

    def find_loader(self, fullname, path=None):
        """We need to override this method for Python 3.x.
        """
        loader, portions = super().find_loader(fullname, path)
        if loader is None:
            pathname = fullname.replace(".", "\\")
            for s in self._suffixes:
                if (pathname + s) in self._files:
                    return self, []
            return None, []
        return loader, portions

    def find_module(self, fullname, path=None):
        result = zipimport.zipimporter.find_module(self, fullname, path)
        if result:
            return result
        fullname = fullname.replace(".", "\\")
        for s in self._suffixes:
            if (fullname + s) in self._files:
                return self
        return None

    def find_spec(self, name, path=None):
        module = self.find_module(name, path)
        if module is not None:
            return util.spec_from_loader(name, module)
        else:
            return None

    def load_module(self, fullname):
        verbose = _memimporter.get_verbose_flag()

        if fullname in sys.modules:
            mod = sys.modules[fullname]
            if verbose:
                sys.stderr.write(
                    "import %s # previously loaded from zipfile %s\n"
                    % (fullname, self.archive))
            return mod
        try:
            return zipimport.zipimporter.load_module(self, fullname)
        except ImportError as err:
            if verbose:
                sys.stderr.write("error loading %s: %s\n"% (fullname, err))

            if sys.version_info >= (3, 0):
                # name of initfunction
                initname = "PyInit_" + fullname.split(".")[-1]
            else:
                # name of initfunction
                initname = "init" + fullname.split(".")[-1]
            filename = fullname.replace(".", "\\")
            if filename in ("pywintypes", "pythoncom"):
                filename = filename + "%d%d" % sys.version_info[:2]
                suffixes = ('.dll',)
            else:
                suffixes = self._suffixes
            for s in suffixes:
                path = filename + s
                if path in self._files:
                    if verbose > 1:
                        sys.stderr.write("# found %s in zipfile %s\n"
                                         % (path, self.archive))
                    spec = util.find_spec(fullname, path)
                    mod = _memimporter.import_module(fullname, path,
                                                     initname,
                                                     self.get_data, spec)
                    mod.__file__ = "%s\\%s" % (self.archive, path)
                    mod.__loader__ = self
                    mod.__memimported__ = True
                    sys.modules[fullname] = mod
                    if verbose:
                        sys.stderr.write("import %s # loaded from zipfile %s\n"
                                         % (fullname, mod.__file__))
                    return mod
            raise zipimport.ZipImportError("can't find module %s" % fullname) from err

    if sys.version_info >= (3, 10):
        def create_module(self, spec):
            mod =  super().create_module(spec)
            if mod is None:
                verbose = _memimporter.get_verbose_flag()
                fullname = spec.name

                filename = fullname.replace(".", "\\")
                suffixes = self._suffixes
                initname = "PyInit_" + fullname.split(".")[-1]

                for s in suffixes:
                    path = filename + s
                    if path in self._files:
                        if verbose > 1:
                            sys.stderr.write("# found %s in zipfile %s\n"
                                            % (path, self.archive))
                        mod = _memimporter.import_module(fullname, path,
                                                        initname,
                                                        self.get_data, spec)
                        mod.__file__ = "%s\\%s" % (self.archive, path)
                        mod.__loader__ = self
                        mod.__memimported__ = True
                        if verbose:
                            sys.stderr.write("import %s # loaded from zipfile %s\n"
                                            % (fullname, mod.__file__))
                        return mod
                # raise zipimport.ZipImportError("can't find module %s" % fullname)

    if sys.version_info >= (3, 10):
        def exec_module(self, module):
            if hasattr(module, '__memimported__'):
                pass
            else:
                super().exec_module(module)

    def __repr__(self):
        return "<%s object %r>" % (self.__class__.__name__, self.archive)

def install():
    "Install the zipextimporter"
    sys.path_hooks.insert(0, ZipExtensionImporter)
    # Not sure if this is needed...
    sys.path_importer_cache.clear()
    ## # Not sure if this is needed...
    ## import importlib
    ## importlib.invalidate_caches()


login.py
import sys
from PyQt5 import QtWidgets, QtGui
from PyQt5.QtWidgets import QApplication, QMainWindow
from PyQt5.QtGui import QIcon
import os

def window():
    app = QApplication(sys.argv)
    win = QMainWindow()
    win.setGeometry(500, 500, 500, 500)
    win.setWindowTitle("PatientNama - Login")
    win.setWindowIcon(QIcon("Me.jpg"))
    win.setToolTip("PatientNama - Patient Record Management APP")
    win.setStyleSheet("background-color:#222021;border:0px solid #48494B;border-radius:5px;color:white;padding:0px;margin-left:5px;")
    
    a=200
    bg_image = QtWidgets.QLabel(win)
    bg_image.setGeometry(580, 0, a, a)
    bg_image.setPixmap(QtGui.QPixmap("Me.jpg").scaled(a, a))
    bg_image.resize(250,250)
    
    def clicked(self):
        print("Button Clicked.")
        if lbl_username1.text()=="admin" and lbl_password1.text()=="12345":
            os.system('python main.py')
            lbl_message.setStyleSheet("background-color: lightgreen;color:black;")
            lbl_message.setText("Access Allowed")
        else:
            lbl_message.setStyleSheet("background-color: red")
            lbl_message.setText("Invalid Username / Password")
            
    

    lbl_username = QtWidgets.QLabel(win)
    lbl_username.setText("Enter Username:")
    lbl_username.move(600, 300)
    lbl_username1 = QtWidgets.QLineEdit(win)
    lbl_username1.move(700, 300)

    lbl_password = QtWidgets.QLabel(win)
    lbl_password.setText("Enter Password:")
    lbl_password.move(600, 350)
    lbl_password1 = QtWidgets.QLineEdit(win)
    lbl_password1.move(700, 350)

    btn_save = QtWidgets.QPushButton(win)
    btn_save.setText('Login')
    btn_save.clicked.connect(clicked)
    btn_save.move(700, 400)
    btn_save.setStyleSheet("background-color: #3b5998; color: white; border-radius: 5px; padding: 5px;")

    lbl_message = QtWidgets.QLabel(win)
    lbl_message.setText("")
    lbl_message.move(600,230)
    lbl_message.resize(200,50)
    lbl_message.setStyleSheet("font-weight: bold;")

    # Add CSS to labels and line edits
    lbl_username.setStyleSheet("font-weight: bold;")
    lbl_password.setStyleSheet("font-weight: bold;")
    lbl_username1.setStyleSheet("border: 1px solid black; border-radius: 5px; padding: 5px;")
    lbl_password1.setStyleSheet("border: 1px solid black; border-radius: 5px; padding: 5px;")

    win.show()
    sys.exit(app.exec_())


if __name__ == "__main__":
    window()
